references/recognition/train_pytorch.py

```diff
@@ -190,88 +122,123 @@ def evaluate(model, val_loader, batch_transforms, val_metric, amp=False, log=Non
 
 
 def main(args):
+    """
+    Args:
+        rank (int): device id to put the model on
+        world_size (int): number of processes participating in the job
+        args: other arguments passed through the CLI
+    """
+    # Detect distributed setup
+    world_size = int(os.environ.get("WORLD_SIZE", 1))
```

Comment:
world_size = dist.get_world_size() ?

Reply:
Additional I think a env var `DOCTR_DISABLE_DDP` would be good here to allow users with multi gpu system to run on a single gpu otherwise it will always default to ddp -> the env var needs to be documented in the readme section